addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpiderAPI.java

```diff
@@ -0,0 +1,194 @@
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    public ClientSpiderAPI(ExtensionClientIntegration extension) {
+        this.extension = extension;
+
+        // Register the actions
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_START_SCAN,
+                        null,
+                        new String[] {
+                            PARAM_URL,
+                            PARAM_MAX_CHILDREN,
+                            PARAM_RECURSE,
+                            PARAM_CONTEXT_NAME,
+                            PARAM_SUBTREE_ONLY
+                        }));
+
+        this.addApiAction(new ApiAction(ACTION_STOP_SCAN, null, new String[] {PARAM_SCAN_ID}));
+
+        // Register the views
+        this.addApiView(new ApiView(VIEW_STATUS, null, new String[] {PARAM_SCAN_ID}));
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        LOGGER.debug("Request for handleApiAction: {} (params: {})", name, params);
+
+        ClientSpider scan;
+        int maxChildren = -1;
+        Context context = null;
+
+        switch (name) {
+            case ACTION_START_SCAN:
+                // The action is to start a new Scan
+                String url = ApiUtils.getOptionalStringParam(params, PARAM_URL);
+                if (params.containsKey(PARAM_MAX_CHILDREN)) {
+                    String maxChildrenStr = params.getString(PARAM_MAX_CHILDREN);
+                    if (maxChildrenStr != null && !maxChildrenStr.isEmpty()) {
+                        try {
+                            maxChildren = Integer.parseInt(maxChildrenStr);
+                        } catch (NumberFormatException e) {
+                            throw new ApiException(
+                                    ApiException.Type.ILLEGAL_PARAMETER, PARAM_MAX_CHILDREN);
+                        }
+                    }
+                }
+                if (params.containsKey(PARAM_CONTEXT_NAME)) {
+                    String contextName = params.getString(PARAM_CONTEXT_NAME);
+                    if (!contextName.isEmpty()) {
+                        context = ApiUtils.getContextByName(contextName);
+                    }
+                }
+
+                ClientOptions options = this.extension.getClientParam();
+                options.setMaxChildren(maxChildren);
+
+                try {
+                    int scanId =
+                            extension.startScan(
+                                    url,
+                                    options,
+                                    context,
+                                    null,
+                                    getParam(params, PARAM_SUBTREE_ONLY, false));
+                    return new ApiResponseElement(name, Integer.toString(scanId));
+                } catch (URIException e) {
+                    throw new ApiException(ApiException.Type.BAD_ACTION);
```

Comment:
If it fails to handle the `url` I guess that's bad param.

Should probably only try to run the spider if there's a url or context, otherwise there's no target.